src/safeposix/syscalls/fs_calls.rs

```diff
@@ -846,29 +934,87 @@ impl Cage {
+    /// a component of path prefix does not exist; or the file
+    /// named by oldpath does not exist.
+    /// * `EISDIR` - When the unlinking is done on a directory
+    ///
+    /// ### Panics
+    ///
+    /// * If the parent inode does not exist in the inode table, causing
+    ///   unwrap() to panic.
+    ///
+    /// For more detailed description of all the commands and return values, see
+    /// [unlink(2)](https://man7.org/linux/man-pages/man2/unlink.2.html)
     pub fn unlink_syscall(&self, path: &str) -> i32 {
+        // Return an error if the provided path is empty
         if path.len() == 0 {
-            return syscall_error(Errno::ENOENT, "unmknod", "given oldpath was null");
+            return syscall_error(Errno::ENOENT, "unlink", "given path was null");
         }
+        // Retrieve the absolute path from the root directory for the given path.
+        // The absolute path is then used to validate directory paths while navigating
+        // through subdirectories.
         let truepath = normpath(convpath(path), self);
 
+        // Walk through the absolute path which returns a tuple consisting of inode
+        // number of file (if it exists), and inode number of parent (if it exists)
         match metawalkandparent(truepath.as_path()) {
-            //If the file does not exist
+            // Return an error if the given file does not exist
             (None, ..) => syscall_error(Errno::ENOENT, "unlink", "path does not exist"),
 
-            //If the file exists but has no parent, it's the root directory
+            // If the file exists but has no parent, it's the root directory
+            // No unlinking is done on the root, and an error is returned
             (Some(_), None) => {
                 syscall_error(Errno::EISDIR, "unlink", "cannot unlink root directory")
             }
 
-            //If both the file and the parent directory exists
+            // If both the file and the parent directory exists
             (Some(inodenum), Some(parentinodenum)) => {
+                // Get the mutable instance of the file from the Inode table
                 let mut inodeobj = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
 
+                // For the inode object, we update 4 parameters:
+                // `reference count`: refers to the active file descriptors pointing to the
+                // file.
+                // `link count`: refers to the number of hard links pointing to the file.
+                // linkcount is decremented by 1 for all inode types except "Dir" type.
+                // `file object`: indicates whether the inode being unlinked has an associated
+                // file object. This is relevant for managing the physical deletion of the file
+                // data from the filesystem. It is only "true" for "File" type inode.
+                // `log`: indicates whether the FileMetaData will be updated for the inode.
```

Comment:
This whole structure is a bit weird.  Since we're not refactoring, I think it's okay to leave it.

Reply:
I just wanted to explain the different parameters in the form of points.